### PR TITLE
Rollback changes to view constructors to reduce the number of instantiations

### PR DESCRIPTION
**Pull request by dalg24**
Temporary fix for #6563

Revert merge of #6536 and add a test capturing the use case that was broken by that change

### CODE REVIEW

File: core/unit_test/TestViewEmptyRuntimeUnmanaged.hpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  Kokkos::View<T*> v1(&d, 0);
+  Kokkos::View<T*> w1(nullptr, 0);
+  Kokkos::View<T*> x1(0, 0);
+  Kokkos::View<T*> y1(NULL, 0);
+
+  Kokkos::View<T**> u2(p, 0, 0);
+  Kokkos::View<T**> v2(&d, 0, 0);
+  Kokkos::View<T**> w2(nullptr, 0, 0);
+  Kokkos::View<T**> x2(0, 0, 0);
+  Kokkos::View<T**> y2(NULL, 0, 0);
```
Comment (masterleinad) [inline core/unit_test/TestViewEmptyRuntimeUnmanaged.hpp:44]: Don't you get unused-type warnings for all of these?

Comment (dalg24) [reply core/unit_test/TestViewEmptyRuntimeUnmanaged.hpp:44]: Oversight

**Review by masterleinad — APPROVED** [review]
I think rolling back is the right step.

**Comment by dalg24** [comment]
(Fixed silly mistake typo `test[_empty]_view_runtime_unmanaged` and squashed into initial commit)

File: core/unit_test/TestViewEmptyRuntimeUnmanaged.hpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  (void)Kokkos::View<T**>(&d, 0, 0);
+  (void)Kokkos::View<T**>(nullptr, 0, 0);
+  (void)Kokkos::View<T**>(NULL, 0, 0);
+  (void)Kokkos::View<T**>(0, 0, 0);
+}
+
+TEST(TEST_CATEGORY, view_empty_runtime_unmanaged) {
+  test_empty_view_runtime_unmanaged<float>();
+  test_empty_view_runtime_unmanaged<int>();
+  test_empty_view_runtime_unmanaged<char>();
```
Comment (crtrott) [inline core/unit_test/TestViewEmptyRuntimeUnmanaged.hpp:50]: lets test const char

Comment (crtrott) [inline core/unit_test/TestViewEmptyRuntimeUnmanaged.hpp:50]:
```suggestion
  test_empty_view_runtime_unmanaged<char>();
  test_empty_view_runtime_unmanaged<const char>();
```

File: core/unit_test/TestViewEmptyRuntimeUnmanaged.hpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  (void)Kokkos::View<T**>(p, 0, 0);
+  (void)Kokkos::View<T**>(&d, 0, 0);
+  (void)Kokkos::View<T**>(nullptr, 0, 0);
+  (void)Kokkos::View<T**>(NULL, 0, 0);
+  (void)Kokkos::View<T**>(0, 0, 0);
+}
+
+TEST(TEST_CATEGORY, view_empty_runtime_unmanaged) {
+  test_empty_view_runtime_unmanaged<float>();
```
Comment (crtrott) [inline core/unit_test/TestViewEmptyRuntimeUnmanaged.hpp:48]:
```suggestion
  test_empty_view_runtime_unmanaged<float>();
  test_empty_view_runtime_unmanaged<const double>();
```

**Review by crtrott — APPROVED** [review]

**Comment by dalg24** [comment]
Ignoring OpenMPTarget failure
```
48: [ RUN      ] simd.device_math_ops
48: "PluginInterface" error: Failure to synchronize stream (nil): Error in cuStreamSynchronize: an illegal memory access was encountered
48: Libomptarget error: Consult https://openmp.llvm.org/design/Runtimes.html for debugging options.
48: Kokkos_OpenMPTarget_ParallelFor_Range.hpp:54:1: Libomptarget fatal error 1: failure of target construct while offloading is mandatory
48/48 Test #48: Kokkos_UnitTest_SIMD ........................Subprocess aborted***Exception: 117.84 sec

**Pull request merged**